lib/_http_incoming.js

```diff
@@ -63,6 +60,9 @@ util.inherits(IncomingMessage, Stream.Readable);
 
 exports.IncomingMessage = IncomingMessage;
 
+IncomingMessage.prototype.__defineGetter__('client', util.deprecate(function() {
```

Comment:
I think we prefer `Object.defineProperty` ([like so](https://github.com/iojs/io.js/blob/master/lib/_http_outgoing.js#L401-L405), with a `util.deprecate`'d getter function as you have here), but otherwise this looks good!